### PR TITLE
initial commit google cloud function

### LINKED ISSUE #39: Add the source code for a google cloud function that is a safeguard

### PR DESCRIPTION
**Pull request by gtfisher**
Signed-off-by: gtfisher <grant@chaosiq.io>

closes #39

### CODE REVIEW

File: google-cloud-function/main.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import datetime
+import json
+import sys
+
+# from flask import jsonify
```
Comment (dmartin35) [inline google-cloud-function/main.py]: I would remove this

File: google-cloud-function/main.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    print('hour_in_range called')
+
+    print(f'request is: {request}')
+    if request.args and 'lower' in request.args:
+        lower = int(request.args.get('lower'))
+        if 'upper' in request.args:
+            upper = int(request.args.get('upper'))
+            return get_json_result(hour_is_in_range(lower, upper))
+
+    INDEX_TEMPLATE = """
```
Comment (dmartin35) [inline google-cloud-function/main.py]: I would define this outside the function, below the imports

File: google-cloud-function/main.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+def hour_is_in_range(lower, upper):
+    """returns a boolean indicating if the current hour is within bounds
+
+    Parameters:
+    lower (int): lowerbound of range
+    upper (int): upper bound of range
+
+    Returns:
+    bool:True if hte current our is between lower and upper, otherwise false
```
Comment (dmartin35) [inline google-cloud-function/main.py]: typo: hte -> the current

File: google-cloud-function/main.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        <body>
+        <h1>Hour in Range Cloud function</h1>
+
+        <p>This is a google cloud function that can be used for safeguards with
+        ChaosIQ</p>
+        <p>Please add parameters giving lowerand upper bounds:</p>
+        <a href="?lower=8&upper=18">hours in range link</a>
+        </body>
+    """
+    return INDEX_TEMPLATE
```
Comment (dmartin35) [inline google-cloud-function/main.py]: maybe add a comment like: for accessing page without arguments, shows the page with usage

File: google-cloud-function/main.py (hunk 1 above)
Comment (gtfisher) [reply google-cloud-function/main.py]: I had intended to, I forgot it was in there

done

**Pull request merged**